src/kohlrahbi/__init__.py

```diff
@@ -27,22 +27,26 @@
 
 
 # pylint:disable=anomalous-backslash-in-string
-def get_valid_pruefis(list_of_pruefis: list[str], all_known_pruefis: Optional[list[str]] = None) -> list[str]:
+def get_valid_pruefis(
+    pruefi_to_file_mapping: dict[str, Union[str, None]], all_known_pruefis: Optional[list[str]] = None
+) -> dict[str, Union[str, None]]:
     """
-    This function returns a new list with only those pruefis which match the pruefi_pattern r"^[1-9]\d{4}$".
+    This function returns a dictionary with only those pruefi/filename - combinations
+    which match the pruefi_pattern r"^[1-9]\d{4}$".
     It also supports unix wildcards like '*' and '?' iff a list of known pruefis is given.
     E.g. '11*' for all pruefis starting with '11' or '*01' for all pruefis ending with '01'.
     """
-    result: set[str] = set()
+    result: dict[str, Union[str, None]] = {}
 
-    for pruefi in list_of_pruefis:
+    for pruefi in pruefi_to_file_mapping:
         if ("*" in pruefi or "?" in pruefi) and all_known_pruefis:
             filtered_pruefis = fnmatch.filter(all_known_pruefis, pruefi)
-            result = result.union(filtered_pruefis)
+            for filtered_pruefi in filtered_pruefis:
+                result.update({filtered_pruefi: pruefi_to_file_mapping.get(filtered_pruefi, None)})
```

Comment:
```suggestion
                if filtered_pruefi in prufi_to_file_mapping:
                    result.update({filtered_pruefi: pruefi_to_file_mapping[filtered_pruefi]})
```
könnte man so nicht die None-Werte ausschließen? aber ich versteh imme rnoch nicht, warum wir sie am ende brauchen